src/dgp/kernels.py

```diff
@@ -84,7 +97,7 @@ def A(dx: Float[Array, "N D"], dy: Float[Array, "N D"]) -> Float[Array, "N N"]:
     )
 
     def B(dx: Float[Array, "N D"], y: Float[Array, "M 1"]) -> Float[Array, "N M"]:
-        batched_matrix = ddx_cov(dx, y)
+        batched_matrix = -ddx_cov(dx, y)
```

Comment:
In the interest of keeping the codebase a little more general, how about using an optional argument to `derivative_cov_func` such as `negative_derivative: bool=False`, where users could indicate if they have observations of the derivative or negative gradient of the function?